code/game/objects/objs.dm

```diff
@@ -33,7 +33,12 @@
 	/// Custom fire overlay icon
 	var/custom_fire_overlay
 
-	var/renamedByPlayer = FALSE //set when a player uses a pen on a renamable object
+	///set when a player changes the name or desc of a renamable object
+	var/renamedByPlayer = FALSE
+	///set when a player changes the name of a renamable object
+	var/name_before_player_changed
+	///set when a player changes the desc of a renamable object
+	var/desc_before_player_changed
 
```

Comment:
OHHHHH it hurts, but I don't know how to fix it

Reply:
Also any reason we can't make renamedByPlayer a flag or something?

Reply:
The only reason I'm aware of is that I don't know how to do that

Reply:
poke me on discord if you want help with it, there's a page on the wiki about bitflags that might help too

Reply:
A component might be better, but the overhead might be worse.

Reply:
It wouldn't be too bad would it? People don't really rename that many things. It would just require converting the system to use signals

Reply:
`/atom/proc/update_name` and `/atom/proc/update_desc` would be convenient for refactoring reagent containers so I don't need to cram all of my name/desc updating code under `/atom/proc/update_icon` as usual.

Reply:
So I figured out how to make renamedByPlayer a bitflag but making those procs with components and signals is still beyond me at the moment

Reply:
if ya need help feel free to ask on discord, people would be more then willing to help

Reply:
@LemonInTheDark @TemporalOroboros If you don't mind and care could you have a look at my shit attempt to do what you've described here and tell me what I've done wrong?